From 783a4996ce0ac64b6e71a640ac4163733d1e7c54 Mon Sep 17 00:00:00 2001
From: Denis Jagoudel <denis.jagoudel@ariadnext.com>
Date: Thu, 8 Dec 2016 17:18:37 +0100
Subject: [PATCH] Patch follow redirect

---
 lib/ApiClient.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/ApiClient.php b/lib/ApiClient.php
index 8c770e0..e5d23cf 100644
--- a/lib/ApiClient.php
+++ b/lib/ApiClient.php
@@ -214,7 +214,7 @@ class ApiClient
             throw new ApiException('Method ' . $method . ' is not recognized.');
         }
         curl_setopt($curl, CURLOPT_URL, $url);
-        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, false);
+        curl_setopt($curl, CURLOPT_FOLLOWLOCATION, false);
 
         // Set user agent
         curl_setopt($curl, CURLOPT_USERAGENT, $this->config->getUserAgent());
@@ -259,7 +259,7 @@ class ApiClient
             $exception = new ApiException($error_message, 0, null, null);
             $exception->setResponseObject($response_info);
             throw $exception;
-        } elseif ($response_info['http_code'] >= 200 && $response_info['http_code'] <= 299) {
+        } elseif (($response_info['http_code'] >= 200 && $response_info['http_code'] <= 299) || ($response_info['http_code'] == 303)) {
             // return raw body if response is a file
             if ($responseType == '\SplFileObject' || $responseType == 'string') {
                 return array($http_body, $response_info['http_code'], $http_header);
-- 
2.9.3 (Apple Git-75)

